server/lib/elastic-search/search.ts

```diff
@@ -118,58 +150,73 @@ const buildQuery = (
+  const { query, fields, indexes } = buildQuery(searchTerm, requestedIndexes, user, account, host);
 
   // Due to permissions, we may end up searching on no index at all (e.g. trying to search for comments while unauthenticated)
   if (indexes.size === 0) {
     return null;
   }
 
-  return client.search({
-    /* eslint-disable camelcase */
-    index: Array.from(indexes).join(','),
-    body: {
-      size: 0, // We don't need hits at the top level
-      query,
-      // Aggregate results by index, keeping only `limit` top hits per index
-      aggs: {
-        by_index: {
-          terms: {
-            field: '_index',
-            size: indexes.size, // Make sure we get all indexes
-          },
-          aggs: {
-            top_hits_by_index: {
-              top_hits: {
-                size: limit,
-                _source: {
-                  // We only need to retrieve the IDs, the rest will be fetched by the loaders
-                  includes: ['id', 'uuid'],
-                },
-                highlight: {
-                  pre_tags: ['<em>'],
-                  post_tags: ['</em>'],
-                  fragment_size: 150,
-                  number_of_fragments: 3,
-                  fields: Array.from(fields).reduce((acc, field) => {
-                    acc[field] = {};
-                    return acc;
-                  }, {}),
+  try {
+    return await client.search({
+      /* eslint-disable camelcase */
+      timeout: `${timeoutInSeconds}s`,
+      index: Array.from(indexes).join(','),
+      body: {
+        size: 0, // We don't need hits at the top level
+        query,
+        // Aggregate results by index, keeping only `limit` top hits per index
+        aggs: {
+          by_index: {
+            terms: {
+              field: '_index',
+              size: indexes.size, // Make sure we get all indexes
+            },
+            aggs: {
+              top_hits_by_index: {
+                top_hits: {
+                  size: limit,
+                  _source: {
+                    // We only need to retrieve the IDs, the rest will be fetched by the loaders
+                    includes: ['id', 'uuid'],
+                  },
+                  highlight: {
+                    pre_tags: ['<em>'],
+                    post_tags: ['</em>'],
+                    fragment_size: 150,
+                    number_of_fragments: 3,
+                    fields: Array.from(fields).reduce((acc, field) => {
+                      acc[field] = {};
+                      return acc;
+                    }, {}),
+                  },
                 },
               },
             },
           },
         },
       },
-    },
-    /* eslint-enable camelcase */
-  });
+      /* eslint-enable camelcase */
+    });
+  } catch (e) {
+    reportErrorToSentry(e, { user, extra: { requestedIndexes, searchTerm, limit, account, host } });
+    throw new Error('The search query failed, please try again later');
+  }
```

Comment:
Added some error handling to make sure we don't expose ES errors in the frontend